scripts/quests/ahtUrhgan/Saga_of_the_Skyserpent.lua

```diff
@@ -128,18 +126,24 @@ quest.sections =
             ['Biyaada'] = quest:event(279),
         },
 
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Fari-Wari'] = quest:event(833)
+        },
+
+    },
+
+    {
+        check = function(player, status, vars)
+            return
+                status == QUEST_ACCEPTED and
+                vars.Prog == 2 and
+                quest:getVar(player, 'Timer') <= VanadielUniqueDay()
+        end,
+
         [xi.zone.AHT_URHGAN_WHITEGATE] =
         {
             ['Fari-Wari'] = quest:progressEvent(825, { text_table = 0 }),
-            {
-                onTrigger = function(player, npc)
-                    if quest:getVar(player, 'Stage') < VanadielUniqueDay() then
-                        return quest:progressEvent(825, { text_table = 0 })
-                    else
-                        return quest:event(833)
```

Comment:
we're removing this extra dialogue here. Or is this their default dialogue? If so it should be in DefaultActions.lua

Reply:
I believe it could be their default. I moved it into this check:

check = function(player, status, vars)
            return status == QUEST_ACCEPTED and vars.Prog == 2

because it wasn't being called before

Reply:
I think a quick test would do us great! If it is, it'll go in sciprts/zones/Aht_Urghan_Whitegate/DefaultActions.lua
If it's not, we should leave it in here